File: Ruby/167. Two Sum II - Input Array Is Sorted.rb
# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
    a, b = numbers.size - 1, 0
    while numbers[a] + numbers[b] != target do
        if numbers[a] + numbers[b] > target then
            a -= 1
        else
            b += 1
        end
    end
    return [b + 1, a + 1]
end

File: PHP/167. Two Sum II - Input Array Is Sorted.php
class Solution {

    /**
     * @param Integer[] $numbers
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($numbers, $target) {
        $a = count($numbers) - 1;
        $b = 0;
        while($numbers[$a] + $numbers[$b] != $target){
            if($numbers[$a] + $numbers[$b] > $target){ $a--; }
            else{ $b ++; }
        }
        return [$b + 1, $a + 1];
    }
}
